test/jdk/java/lang/reflect/code/TestSwitchExpressionOp.java

```diff
@@ -15,112 +16,198 @@
  */
 public class TestSwitchExpressionOp {
 
-    // TODO more testing
-    //  cover cases where MatchException will be thrown
+    @Test
```

Comment:
Shouldn't we add similar tests in the compiler (with the `@IR` annotation) to check that the model generated by javac contains the expected conversions?

Reply:
You mean a test similar to `caseConstantOtherKindsOfExpr` ?

Reply:
> You mean a test similar to `caseConstantOtherKindsOfExpr` ?

At least something like that yes.